### PR TITLE
[AST] emitLetToVarNoteIfSimple should also check if the method/accessor is explicitly non-mutating

### PR DESCRIPTION
**Pull request by theblixguy**
`emitLetToVarNoteIfSimple()` emits a diagnostic with a fix-it to add `mutating` to a method or accessor. However, it's possible that the method or accessor is explicitly `nonmutating`. In that case, we end up adding the `mutating` after `nonmutating`, leading to another error: `Method must not be declared both nonmutating and mutating` (if we have a method) or `Expected 'get', 'set', 'willSet', or 'didSet' keyword to start an accessor definition` (if we have an accessor, like a `set`).

We should check whether we have an explicit `nonmutating` attribute and instead of inserting the fix-it, we should replace `nonmutating` with `mutating`.

(I noticed when I was writing some tests!)

### CODE REVIEW

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Comment by theblixguy** [comment]
@swift-ci please smoke test

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -5683,11 +5683,16 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
...
+
       auto &d = getASTContext().Diags;
-      d.diagnose(FD->getFuncLoc(), diag::change_to_mutating,
-                 isa<AccessorDecl>(FD))
-       .fixItInsert(FD->getFuncLoc(), "mutating ");
+      auto diags = d.diagnose(FD->getFuncLoc(), diag::change_to_mutating,
+                              isa<AccessorDecl>(FD));
+      if (auto nonmutatingAttr =
+              FD->getAttrs().getAttribute<NonMutatingAttr>()) {
+        diags.fixItReplace(nonmutatingAttr->getLocation(), "mutating ");
```
Comment (jrose-apple) [inline lib/AST/Decl.cpp]: This one shouldn't have the trailing space (even though the diagnostic engine will handle it, let's be pedantically correct).

**Review by jrose-apple — APPROVED** [review]

File: lib/AST/Decl.cpp (hunk 1 above)
Comment (theblixguy) [reply lib/AST/Decl.cpp]: Sure, I have done that now!

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Pull request merged**